src/NATS.Client.KeyValueStore/NatsKVOpts.cs

```diff
@@ -41,6 +41,14 @@ public record NatsKVWatchOpts
     /// Async function called when the enumerator reaches the end of data. Return True to break the async enumeration, False to allow the enumeration to continue.
     /// </summary>
     public Func<CancellationToken, ValueTask<bool>>? OnNoData { get; init; }
+
+    /// <summary>
+    /// The revision to start from, if set to 0 (default) this will be ignored.
+    /// <remarks>
+    /// Setting this to a non-zero value will cause the watcher to ignore the values for <see cref="IncludeHistory"/> and <see cref="UpdatesOnly"/>.
+    /// </remarks>
```

Comment:
Is it more consistent/correct to annotate that this option silently overrides other options?  Or should we fail with an Exception that states that two options that are mutually exclusive have been set?

Reply:
Probably whichever pattern we follow in Consumer Creation is the same pattern we would want to follow here

Reply:
On second thought it might be a little abrasive to throw here, since `IncludeHistory` will default to `false` but when setting `ResumeAtRevision` it _must_ be true.  I think it makes sense to ignore `IncludeHistory` when `ResumeAtRevision` is set.

`UpdatesOnly = true` and `ResumeAtRevision > 0` are 2 non-defaults though so I think it makes sense to throw when those are both set.

[Regular Consumer Crerate](https://github.com/nats-io/nats.net.v2/blob/8952bc8bd1a0e60c1848eabd4e4f52326c1de4ef/src/NATS.Client.JetStream/NatsJSContext.Consumers.cs#L203-L225) isn't checking any pre-conditions it looks like, it's taking the Deliver Policy and Start Seq / Start Time separately.  So if the Deliver Policy doesn't match the Server will fail with an error, and an exception will be thrown.

Reply:
`ArgumentException` is consistent with the library

Reply:
Calling GetKeys with the resume parameter might be a valid use case. I agree that including history is overridden due to the method overload. However, considering the following points, I suggest keeping it in the options:

1) Cross-cutting functionality: resume seems to be applicable to multiple scenarios and not specific to a single use case. Including it in the options would make it more reusable and accessible across different parts of the codebase.

2) Consistency with other NATS clients: It's important to maintain a certain degree of consistency with other client implementations, unless there are .NET specific differences.

3) Impact on other clients: Altering the method signature or introducing overloads would require updating other client implementations as well. This can be a significant effort, and there should be a strong justification for making such changes unless we can say this is a .NET specific difference.

Reply:
I was checking Go and JS implementations https://github.com/nats-io/nats.go/pull/1489/files https://github.com/nats-io/nats.deno/pull/651/files .. so we're not that consistent after all😄

I guess we can add overloads if you think it's going to be a better developer experience. it just feels like a less generic solution to me. wdyt @niklasfp @caleblloyd

Reply:
I am also in favor of leaving it in Options and adding Validation.

If the Validation needs to run from multiple callers, we could add a `internal void ApplyOptsToConsumerConfig(ConsumerConfig config)` method on the Options, which performs validation, then mutates a ConsumerConfig?

We do something similar [in NatsTlsOpts](https://github.com/nats-io/nats.net.v2/blob/df478c27425a12c786f604fd9da8b4f9c83e71a3/src/NATS.Client.Core/NatsTlsOpts.cs#L163-L172) - where validation is performed at the time of use